drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+       * in memory.
+       */
+
+      dest = gd55_read_cache(priv, sector);
+
+      /* Erase the block containing this sector if it is not already erased.
+       * The erased indicated will be cleared when the data from the erase
+       * sector is read into the cache and set here when we erase the sector.
+       */
+
+      if (!IS_ERASED(priv))
+        {
+          off_t esectno  = sector >>
+                           (GD55_SECTOR_SHIFT - GD55_SECTOR512_SHIFT);
+          finfo("sector: %jd esectno: %jd\n",
+                (intmax_t)sector, (intmax_t)esectno);
+
+          ret = gd55_erase_sector(priv, esectno);
+          if (ret < 0)
+            {
+              ferr("ERROR: gd55_erase_sector failed: %d\n", ret);
+              return ret;
+            }
+
+          SET_ERASED(priv);
+        }
+
+      /* Copy the new sector data into cached erase block */
+
+      memcpy(dest, buffer, GD55_SECTOR512_SIZE);
+      SET_DIRTY(priv);
+
+      /* Set up for the next 512 byte sector */
+
+      finfo("address: %08jx nbytes: %d 0x%04" PRIx32 "\n",
+            (intmax_t)(sector << GD55_SECTOR512_SHIFT),
+            GD55_SECTOR512_SIZE,
+            *(FAR uint32_t *)buffer);
+      buffer += GD55_SECTOR512_SIZE;
+      sector++;
+    }
+
+  /* Flush the last erase block left in the cache */
+
+  return gd55_flush_cache(priv);
+}
+#endif /* CONFIG_MTD_GD55_SECTOR512 */
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: gd55_initialize
+ *
+ * Description:
+ *   Create an initialize MTD device instance.
+ *
+ *   MTD devices are not registered in the file system, but are created as
+ *   instances that can be bound to other functions (such as a block or
+ *   character driver front end).
+ *
+ * Input Parameters:
+ *   qspi       - a reference to the qspi device to initialize
+ *   unprotect  - if true, unprotect the device
+ *
+ * Returned Value:
+ *   Success (OK) or fail (negated error code)
+ ****************************************************************************/
+
+FAR struct mtd_dev_s *gd55_initialize(FAR struct qspi_dev_s *qspi,
+                                         bool unprotect)
+{
+  FAR struct gd55_dev_s *dev;
+  int ret;
+  uint8_t status;
+  uint16_t config;
+
+  DEBUGASSERT(qspi != NULL);
+
+  /* Allocate a state structure (we allocate the structure instead of using
+   * a fixed, static allocation so that we can handle multiple FLASH devices.
+   * The current implementation would handle only one FLASH part per QuadSPI
+   * device (only because of the QSPIDEV_FLASH(0) definition) and so would
+   * have to be extended to handle multiple FLASH parts on the same QuadSPI
+   * bus.
+   */
+
+  dev = kmm_zalloc(sizeof(*dev));
+
```

Comment:
remove blank line.

Reply:
OK - not picked up by checkpatch.sh, but agree. Thanka.

Reply:
done